src/_utils/useOverflowHiddenOnDocument/index.tsx

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react'
+
+const DEFAULT_OVERFLOW_VALUE = 'visible'
+const OVERFLOW_HIDDEN_VALUE = 'hidden'
+
+/**
+ * This custom hook will remove the overflow visible from the document
+ * It's useful for modal or any screen that wants to handle their own scrollbar
+ * WARNING: For now it doesn't handle 2 components using this hooks at the same time
+ */
+export const useOverflowHiddenOnDocument = () => {
```

Comment:
Not sure why we use a hook/useEffect for that.
It's can be a simple helper static function, no need to tight it to the component rendering lifecycle.

Reply:
You're right, I was on my to create custom hooks so I kept going.
I'll create a static function with a condition on being run on client side.

Reply:
We can have something like:
```tsx
function hideOverflowOnDocument() {
  document.querySelector('html').style.overflow = OVERFLOW_HIDDEN_VALUE

  // Return a "clean up/cancel" callback
  return () => {
    document.querySelector('html').style.overflow = DEFAULT_OVERFLOW_VALUE
  }
}

function MyComponent() {
  useEffect(hideOverflowOnDocument)
  // ...
}
```

What do you think?

Reply:
The hook is not extremely useful as a globally shareable hook. I'm afraid we end up with a bunch of confusing barely-reusable global hooks.
I like Simon alternative or you could have a folder-local hook (but if we use it once, why create a separate file/function).

Reply:
How about integrating it _in_ `useFocusTrap`?
I don't have a use case where we want to `useFocusTrap` without preventing a scroll on the document 🤔.

Reply:
> How about integrating it _in_ `useFocusTrap`?
> I don't have a use case where we want to `useFocusTrap` without preventing a scroll on the document 🤔.

I thought of that at first but ended making two separate hooks. I think this is the right approach, I don't see any case where we would have a focusTrap without preventing the scroll on document. I'll add it directly in `usefocusTrap`

Reply:
I removed `useOverflowHiddenOnDocument ` and integrated the logic in useFocusTrap